hs_core/tests/api/rest/test_set_access_rules.py

```diff
@@ -59,11 +73,130 @@ def test_set_access_rules(self):
+                                           keywords=keywords)
+        new_res.metadata.create_element('description', abstract=abstract)
+        res_id = new_res.short_id
+        self.resources_to_delete.append(res_id)
+
+        access_url = "/hsapi/resource/{res_id}/access/".format(res_id=res_id)
+        put_response = self.client.put(access_url, {
+            "privilege": PrivilegeCodes.VIEW,
+            "group_id": self.testGroup.id
+        }, format='json')
+        self.assertEqual("Resource access privileges added.", put_response.data['success'])
+
+        get_response = self.client.get(access_url)
+        self.assertEqual(1, len(get_response.data['users']))
+        self.assertEqual(1, len(get_response.data['groups']))
+
+        delete_response = self.client.delete(access_url + "?group_id=" + str(self.testGroup.id))
+        self.assertEqual("Resource access privileges removed.", delete_response.data['success'])
+
+        get_response = self.client.get(access_url)
+        self.assertEqual(0, len(get_response.data['groups']))
+
+    def test_errors(self):
+        rtype = 'GenericResource'
+        title = 'My Test resource'
+        keywords = ('foo', 'bar')
+        abstract = 'This is a resource used for testing /hsapi/resource/{id}/access/'
+        new_res = resource.create_resource(rtype,
+                                           self.user,
+                                           title,
+                                           keywords=keywords)
+        new_res.metadata.create_element('description', abstract=abstract)
+        res_id = new_res.short_id
+        self.resources_to_delete.append(res_id)
+
+        access_url = "/hsapi/resource/{res_id}/access/".format(res_id=res_id)
+
+        # Empty PUT
+        put_response = self.client.put(access_url, {}, format='json')
+        self.assertEqual(
+            "Request must contain a 'resource' ID as well as a 'user_id' or 'group_id'",
+            put_response.data['error'])
+
+        # Incorrect PUT
+        put_response = self.client.put(access_url, {
+            "user_id": self.secondUser.id,
+            "group_id": self.testGroup.id
+        }, format='json')
+        self.assertEqual(
+            "Request cannot contain both a 'user_id' and a 'group_id' parameter.",
+            put_response.data['error'])
+
+        # Empty DELETE
+        delete_response = self.client.delete(access_url)
+        self.assertEqual(
+            "Request must contain a 'resource' ID as well as a 'user_id' or 'group_id'",
+            delete_response.data['error'])
+
+        # Incorrect DELETE
+        delete_response = self.client.put(access_url + "?user_id=2&group_id=3")
+        self.assertEqual(
+            "Request cannot contain both a 'user_id' and a 'group_id' parameter.",
+            put_response.data['error'])
+
+    def tearDown(self):
+        super(TestSetAccessRules, self).tearDown()
+        self.secondUser.delete()
```

Comment:
I think yes, based on the tearDown method in /hs_core/tests/api/rest/base.py. Am I wrong?